code/espurna/ir.cpp

```diff
@@ -247,7 +247,7 @@ void _irTXLoop() {
         #if IR_USE_RAW
             _ir_sender.sendRaw(_ir_raw, _ir_repeat_size, _ir_freq);
         #else
-            _ir_sender.send(_ir_type, _ir_code, _ir_bits);
+            _ir_sender.send((decode_type_t)_ir_type, _ir_code, _ir_bits);
```

Comment:
Having noticed that... _ir_type should be `int`, since we deal with enum

Reply:
I am not sure I understand you (my English is a little bit rusty :) ). 
The _IRsend::send_ method expects a _decode_type_t_ enum so __ir_type_ as _int_ I don't think is possible. 
To have it as _decode_type_t_  ... well I guess that would be the elegant solution. We then need a helper method to convert the value received from MQTT (line 211) to _decode_type_t_.

Reply:
Hm. So:
- `uint8_t _ir_type` -> `decode_type_t _ir_type`
- `_ir_type = static_cast<decode_type_t>(data.toInt());`
- `static_cast<int>(_ir_type)` for debug print

(I missed the fact that _ir_type never gets changed, I though the reason behind uint8_t it was it added / subtracted / masked somehow)

Reply:
Can you "fix' this pull request ? I tried to edit it but instead of removing the second commit I managed to add a third one :( .

Reply:
Git is weird like that. Done. Thanks!